WIP: Migration integration test

zacshowa opened the pull request:
Closes https://github.com/EspressoSystems/nitro-espresso-integration/issues/162

### This PR:

This is a draft PR to get feedback on current work related to the migration test. Current work revolves around editing the test framework code to give easier access to the Osp and ChallengeManager contract addresses such that we can use them for the integration test.

### This PR does not:

Currently include a integration test

### Key places to review:

Code in espresso_migration_test.go and associated new functions with `NonEspresso` in the name that allow for better accessibility to specific contract bindings that are needed at the top level of the test. 

<!-- ### How to test this PR:  -->
None currently.

<!-- ### Things tested -->
None currently

zacshowa committed:
Add precompiled bindings for a subset of smart contracts to allow standing up a network with non espresso smart contracts

zacshowa committed:
Add new/modified test code to set the Osp and ChallengeManager in the l1 info for use in the integration test.

zacshowa committed:
Commit with current progress for visibility.

zacshowa committed:
Refactor to optimize for code reuse
TODO: use chain config for espresso toggle in builder instead of passing an argument.

zacshowa committed:
Refactor to optimize for code reuse
TODO: use chain config for espresso toggle in builder instead of passing an argument.

zacshowa committed:
remove accidentally included import.

zacshowa committed:
Commit work on test with contract upgrades done.

zacshowa committed:
Commit work on test with contract upgrades done.

zacshowa committed:
Commit current test progress

zacshowa committed:
commit with current progress and some comments surrounding the current issues.

zacshowa committed:
Commit with contract upgrades working correctly.

zacshowa committed:
Add logging to switch_sequencer.go for debugging

zacshowa committed:
Updates to migration test with current progress.

zacshowa committed:
Updates to migration test with current progress.

ImJeremyHe committed:
Espresso sequencer config

ImJeremyHe commented on system_tests/espresso_migration_test.go:
```diff
@@ -0,0 +1,351 @@
+
+	log.Info("Successfully upgraded the challenge manager to point to the new OSP entry.")
+
+	log.Info("Turn on espresso sequencing by changing the builders configs.")
```
I doubt that we can do such a thing. In the `switch sequencer` or `espresso sequencer`, we didn't add any listener to the chain config. That means updating the chain config on L1 won't help.

I think we should turn off the L2 node (which is actually a sequencer here) and run another node to complete the migration.

zacshowa committed:
remove logging from switch_sequencer.go

zacshowa committed:
Update migration test to stop the previous sequencer node, and spin up a new node for espresso sequencing.

zacshowa committed:
Update migration test to stop the previous sequencer node, and spin up a new node for espresso sequencing.

zacshowa committed:
Add comment about current state of migration.

zacshowa committed:
Modify arbosstate.go to allow for upgrade to ArbOS version 35 for the espresso migration

zacshowa committed:
commit current state of E2E test such that transactions are being validated on after being made on the l2 after switching to espresso mode.

zacshowa committed:
Finalized E2E test that passes.

sveitser commented:
@zacshowa this needs to be synced with `integration`.

zacshowa committed:
Sync with integration

zacshowa committed:
Merge changes to espresso_e2e_test.go

zacshowa committed:
Fix erroneous artifacts resulting from syncing with `integration`

zacshowa committed:
update submodule commits to fix compilation issues

zacshowa committed:
remove unnecessary naming of import in Migration test

zacshowa committed:
Update submodule pointer for go-ethereum

zacshowa committed:
run make lint fmt and fix remaining lint issues.

sveitser committed:
Ensure epresso-dev-node image is up to date

- Move Espresso Migration test to espresso E2E tests
- Simplify go test commands

sveitser committed:
fix gotestsum command

zacshowa committed:
remove unused go bindings for non espresso contracts and fix test imports.

zacshowa committed:
Remove unused bridgegen.go

zacshowa committed:
Sync PR branch with integration branch

zacshowa committed:
Commit with current progress fixing migration test after syncing with integration branch

ImJeremyHe committed:
Update go-ethereum

ImJeremyHe committed:
Update submodule contracts

sveitser commented:
Closing because we're testing this with the test-node now.
